File: ga/4omini_result_2024_7/result/1968E_cpp_mini/46_1968E.cpp
use std::io;

fn solve() {    
    let mut n = String::new();
    io::stdin().read_line(&mut n).expect("Failed to read line");
    let n: usize = n.trim().parse().expect("Please type a number!");

    let mut t = n;
    println!("1 1");
    println!("2 1");
    t -= 2;

    if t > 0 {
        println!("{} {}", n, n);
        t -= 1;
    }

    for i in 3..3 + t {
        println!("1 {}", i);
    }
}

fn main() {
    let mut t = String::new();
    io::stdin().read_line(&mut t).expect("Failed to read line");
    let t: usize = t.trim().parse().expect("Please type a number!");

    for _ in 0..t {
        solve();
    }
}

File: ga/glm4_result_2024_6/result/1968E_glm/46_1968E.rs
use std::io::{self, Read};

fn solve(n: i32) {
    let t = n - 2;
    println!("1 1");
    println!("2 1");
    if t > 0 {
        println!("{} {}", n, n);
        println!("1 {}", n + 1);
        for i in 4..=n {
            println!("1 {}", i);
        }
    } else {
        for i in 3..=n {
            println!("1 {}", i);
        }
    }
}

fn main() {
    let mut buffer = String::new();
    io::stdin().read_to_string(&mut buffer).expect("Failed to read from stdin");
    let lines: Vec<&str> = buffer.lines().collect();

    let t = lines[0].parse::<i32>().expect("Failed to parse t");

    for i in 1..=t {
        let n = lines[i].parse::<i32>().expect("Failed to parse n");
        solve(n);
    }
}